pypeit/reduce.py

```diff
@@ -389,9 +389,20 @@ def run(self, ra=None, dec=None, obstime=None, std_trace=None, show_peaks=False,
             msgs.info("Skipping 2nd run of finding objects")
 
         # Assign here -- in case we make another pass to add in missing targets
-        if self.nobj > 0 and self.par['reduce']['slitmask']['assign_obj'] and self.slits.maskdef_designtab is not None:
-            self.slits.assign_maskinfo(self.sobjs_obj, self.get_platescale(None),
-                                       TOLER=self.par['reduce']['slitmask']['obj_toler'])
+        if self.par['reduce']['slitmask']['assign_obj'] and self.slits.maskdef_designtab is not None:
```

Comment:
I somehow missed this on past PRs where this was added (i.e. assign_maskinfo). None of this should be here. This run method for reduce is abstracted to be independent of whether you are reducing Echelle, MultisSlit or IFU data. Likewise for the find_objects method. The find_object_pypeline is then a dummy method that gets overloaded by the instrument specific children. The same philosophy is used for the local_skysub_extract calls, i.e. the mode specific routines are part of the children. 

If you add Multislit specific function calls here with if then statements,  you are violating the philosophy that this abstraction is based on. It is equivalent to have if this spectrograph do this calls all throughout pypeit which we are committed to avoid. 

My suggestion is that these MultSlit specific function calls be added to find_objects_pypeline in the MultiSlit child, which is where anything that is multisit specific needs to be. I can help you do that, with relatively minor changes.